### PR TITLE
Allow `int, floating point` multiplication with singleton types

### LINKED ISSUE #36074: Allow int varref with other numeric operands in a multiplicative expression does not work with singletons
**Description:**
With https://github.com/ballerina-platform/ballerina-lang/pull/35616 we have allowed  `int*float, float*int, int* decimal, decimal*int, float/int, decimal/int, float%int` and `decimal%int`. in multiplicative expression. But it is not working with singleton types. eg:

```bal
5 a = 5;
float b = 2;
float e = a * b; // gives error operator '*' not defined for '5' and 'float'
```

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

Fixes #36074

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by KavinduZoysa** [comment]
@rdulmina,

Do we have test cases like this,
```
type FIVE 5;
type ONE_POINT_FIVE 1.1;

public function main() {
    FIVE five = 5;
    ONE_POINT_FIVE onePointFive = 1.1;
    float _ = five / onePointFive;
}
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 1]
```diff
@@ -1947,24 +1947,44 @@ public BSymbol getArithmeticOpsForTypeSets(OperatorKind opKind, BType lhsType, B
...
+            BType returnType;
             if (compatibleType1.tag < compatibleType2.tag) {
-                return createBinaryOperator(opKind, lhsType, rhsType, compatibleType2);
-            }
-            if (lhsType.isNullable()) {
-                compatibleType1 = BUnionType.create(null, compatibleType1, symTable.nilType);
-                return createBinaryOperator(opKind, lhsType, rhsType, compatibleType1);
+                returnType = compatibleType2;
+            } else {
+                returnType = compatibleType1;
```
Comment (KavinduZoysa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:1959]: Shall we inline this as below,
`BType returnType = compatibleType1.tag < compatibleType2.tag ? compatibleType1 : compatibleType2;`

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 2]
```diff
@@ -1947,24 +1947,44 @@ public BSymbol getArithmeticOpsForTypeSets(OperatorKind opKind, BType lhsType, B
                 compatibleType2 = types.findCompatibleType(rhsType);
             }
 
-            if (types.isBasicNumericType(compatibleType1) && compatibleType1 != compatibleType2) {
+            if (types.isBasicNumericType(compatibleType1) && compatibleType1 != compatibleType2 &&
```
Comment (KRVPerera) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:1950]: shall we move `compatibleType1 != compatibleType2` as first logic here

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 3]
```diff
@@ -1947,24 +1947,44 @@ public BSymbol getArithmeticOpsForTypeSets(OperatorKind opKind, BType lhsType, B
...
+            return createBinaryOperator(opKind, lhsType, rhsType, returnType);
         }
         return symTable.notFoundSymbol;
     }
+    
+    private boolean isIntFloatingPointMultiplication(OperatorKind opKind, BType lhsCompatibleType, 
+                                                     BType rhsCompatibleType) {
+        switch (opKind) {
+            case MUL:
+                return lhsCompatibleType.tag == TypeTags.INT && isFloatingPointType(rhsCompatibleType) ||
```
Comment (KRVPerera) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:1975]: We can extract some variables here

**Review by KRVPerera — APPROVED** [review]
LGTM

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java (hunk 3 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:1975]: Can you mention a sample?

**Comment by rdulmina** [comment]
> @rdulmina,
> 
> Do we have test cases like this,
> 
> ```
> type FIVE 5;
> type ONE_POINT_FIVE 1.1;
> 
> public function main() {
>     FIVE five = 5;
>     ONE_POINT_FIVE onePointFive = 1.1;
>     float _ = five / onePointFive;
> }
> ```
> 
> Also tests related to `const`

Added in [d5c11e3](https://github.com/ballerina-platform/ballerina-lang/pull/36085/commits/d5c11e3778edcfb801906c14b6f3f6f28c9f0695)

**Review by KavinduZoysa — APPROVED** [review]

**Pull request merged**